server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -488,7 +510,7 @@ private int submitCompactionTasks(
       LOG.debugSegments(segmentsToCompact, "Compacting segments");
       // Count the compaction task itself + its sub tasks
       numSubmittedTasks++;
-      numCompactionTasksAndSubtasks += findMaxNumTaskSlotsUsedByOneCompactionTask(config.getTuningConfig());
+      numCompactionTasksAndSubtasks += numCurrentCompactionTasksAndSubtasks;
```

Comment:
These two field names are confusing.

Maybe rename `numCompactionTasksAndSubtasks` to simply `totalTaskSlotsAssigned` or similar, and rename `numCurrentCompactionTasksAndSubtasks` to `slotsRequiredForCurrentTask`.